packages/backend/src/clients/Slack/Slackbot.ts

```diff
@@ -0,0 +1,127 @@
+    router: apiV1Router,
+    scopes: ['links:read', 'links:write'],
+    installationStore: {
+        storeInstallation: createInstallation,
+        fetchInstallation: getInstallation,
+        deleteInstallation,
+    },
+    // Slack only allow https on redirections
+    // When testing locally on http://localhost:3000, replace again https:// with http:// after redirection happens
+    redirectUri: `${(process.env.SITE_URL || '').replace(
+        'http://',
+        'https://',
+    )}/api/v1/slack/oauth_redirect`,
+    installerOptions: {
+        directInstall: true,
+        redirectUriPath: '/slack/oauth_redirect',
+    },
+};
+
+export const receiver = new ExpressReceiver(slackOptions);
+
+apiV1Router.get('/slack/install/:organizationUuid', async (req, res, next) => {
+    try {
+        const options = {
+            redirectUri: slackOptions.redirectUri,
+            scopes: slackOptions.scopes,
+            metadata: { organizationUuid: req.params.organizationUuid },
+        };
+        await receiver.installer.handleInstallPath(
+            req,
+            res,
+            slackOptions.installerOptions,
+            options,
+        );
+    } catch (error) {
+        next(error);
+    }
+});
+
+const app = new App({
+    ...slackOptions,
+    logLevel: LogLevel.INFO,
+    port: process.env.SLACK_PORT || 4000,
+    socketMode: true,
+    appToken: process.env.SLACK_APP_TOKEN,
+});
+
+const unfurl = (event: any, client: any) => {
+    const unfurls = event.links.reduce((acc: any, l: any) => {
+        const { url } = l;
+
+        return {
+            [url]: {
+                blocks: [
+                    {
+                        type: 'header',
+                        text: {
+                            type: 'plain_text',
+                            text: 'Lightdash URL unfurls are not yet available',
+                        },
+                    },
+
+                    /* {
+                        type: 'image',
+                        title: {
+                            type: 'plain_text',
+                            text: 'Public dashboard',
+                        },
+                        block_id: 'image4',
+                        image_url: imgUrl,
+                        alt_text: 'An incredibly cute kitten.',
+                    },
+                    {
+                        type: 'section',
+                        text: {
+                            type: 'mrkdwn',
+                            text: `<${dashboardUrl}|${dashboardName}> `,
+                        },
+                        accessory: {
+                            type: 'button',
+                            text: {
+                                type: 'plain_text',
+                                text: 'Open dashboard',
+                                emoji: true,
+                            },
+                            value: 'view_alternate_1',
+                        },
+                    }, */
```

Comment:
this chunk will be enabled after we deploy headless chrome right? before that we can just test that integration is working?

Reply:
yes, right now there are no images to display. we might also add a few extra blocks

Reply:
nice!